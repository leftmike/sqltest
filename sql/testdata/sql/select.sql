--
-- Test SELECT
--

SELECT 1 AS c1, 2 AS c2;

SELECT 1 + 2 AS c1, 2 + 3 AS c2, 4 * 5 AS c3;

DROP TABLE IF EXISTS tbl1;

CREATE TABLE tbl1 (c1 int, c2 int, c3 int);

INSERT INTO tbl1 VALUES
    (0, 1, 2),
    (3, 4, 5),
    (6, 7, 8),
    (9, 10, 11),
    (12, 13, 14),
    (15, 16, 17),
    (18, 19, 20),
    (21, 22, 23),
    (24, 25, 26),
    (27, 28, 29),
    (30, 31, 32);

SELECT * FROM tbl1;

SELECT tbl1.* FROM tbl1;

SELECT c2 FROM tbl1;

SELECT c3, c1 FROM tbl1;

SELECT c3, c1 FROM tbl1 WHERE c2 < 10;

SELECT c2 FROM tbl1 WHERE c1 > 6 AND c3 <23;

SELECT c2, c3, c1 FROM tbl1 WHERE c1 = 24;

DROP TABLE IF EXISTS tbl2;

CREATE TABLE tbl2 (c1 int, c2 int, c3 int);

INSERT INTO tbl2 VALUES
    (6, 60, 600),
    (7, 70, 700),
    (8, 80, 800),
    (9, 90, 900),
    (10, 100, 1000),
    (11, 110, 1100),
    (12, 120, 1200);

SELECT * FROM tbl2 WHERE c3 = c1 * c2;

SELECT tbl2.* FROM tbl2 WHERE c1 * 10 = c2 AND c2 * 10 = c3;
