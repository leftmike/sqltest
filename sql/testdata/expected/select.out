--
-- Test SELECT
--
SELECT 1 AS c1, 2 AS c2;
   c1 c2
   -- --
 1  1  2
(1 row)
SELECT 1 + 2 AS c1, 2 + 3 AS c2, 4 * 5 AS c3;
   c1 c2 c3
   -- -- --
 1  3  5 20
(1 row)
DROP TABLE IF EXISTS tbl1;
CREATE TABLE tbl1 (c1 int, c2 int, c3 int);
INSERT INTO tbl1 VALUES
    (0, 1, 2),
    (3, 4, 5),
    (6, 7, 8),
    (9, 10, 11),
    (12, 13, 14),
    (15, 16, 17),
    (18, 19, 20),
    (21, 22, 23),
    (24, 25, 26),
    (27, 28, 29),
    (30, 31, 32);
SELECT * FROM tbl1;
    c1 c2 c3
    -- -- --
  1  0  1  2
  2 12 13 14
  3 15 16 17
  4 18 19 20
  5 21 22 23
  6 24 25 26
  7 27 28 29
  8  3  4  5
  9 30 31 32
 10  6  7  8
 11  9 10 11
(11 rows)
SELECT tbl1.* FROM tbl1;
    c1 c2 c3
    -- -- --
  1  0  1  2
  2 12 13 14
  3 15 16 17
  4 18 19 20
  5 21 22 23
  6 24 25 26
  7 27 28 29
  8  3  4  5
  9 30 31 32
 10  6  7  8
 11  9 10 11
(11 rows)
SELECT tbl1.*, c1 + c2 + c3 AS total FROM tbl1;
    c1 c2 c3 total
    -- -- -- -----
  1  0  1  2     3
  2 12 13 14    39
  3 15 16 17    48
  4 18 19 20    57
  5 21 22 23    66
  6 24 25 26    75
  7 27 28 29    84
  8  3  4  5    12
  9 30 31 32    93
 10  6  7  8    21
 11  9 10 11    30
(11 rows)
SELECT c1, c2, c3 FROM tbl1;
    c1 c2 c3
    -- -- --
  1  0  1  2
  2 12 13 14
  3 15 16 17
  4 18 19 20
  5 21 22 23
  6 24 25 26
  7 27 28 29
  8  3  4  5
  9 30 31 32
 10  6  7  8
 11  9 10 11
(11 rows)
SELECT c2, c3, c1 FROM tbl1;
    c2 c3 c1
    -- -- --
  1  1  2  0
  2 10 11  9
  3 13 14 12
  4 16 17 15
  5 19 20 18
  6 22 23 21
  7 25 26 24
  8 28 29 27
  9 31 32 30
 10  4  5  3
 11  7  8  6
(11 rows)
SELECT c2 FROM tbl1;
    c2
    --
  1  1
  2 10
  3 13
  4 16
  5 19
  6 22
  7 25
  8 28
  9 31
 10  4
 11  7
(11 rows)
SELECT c3, c1 FROM tbl1;
    c3 c1
    -- --
  1 11  9
  2 14 12
  3 17 15
  4  2  0
  5 20 18
  6 23 21
  7 26 24
  8 29 27
  9 32 30
 10  5  3
 11  8  6
(11 rows)
SELECT c3, c1 FROM tbl1 WHERE c2 < 10;
   c3 c1
   -- --
 1  2  0
 2  5  3
 3  8  6
(3 rows)
SELECT c2 FROM tbl1 WHERE c1 > 6 AND c3 <23;
   c2
   --
 1 10
 2 13
 3 16
 4 19
(4 rows)
SELECT c2, c3, c1 FROM tbl1 WHERE c1 = 24;
   c2 c3 c1
   -- -- --
 1 25 26 24
(1 row)
DROP TABLE IF EXISTS tbl2;
CREATE TABLE tbl2 (c1 int, c2 int, c3 int);
INSERT INTO tbl2 VALUES
    (6, 60, 600),
    (7, 70, 700),
    (8, 80, 800),
    (9, 90, 900),
    (10, 100, 1000),
    (11, 110, 1100),
    (12, 120, 1200);
SELECT * FROM tbl2 WHERE c3 = c1 * c2;
   c1  c2   c3
   --  --   --
 1 10 100 1000
(1 row)
SELECT tbl2.* FROM tbl2 WHERE c1 * 10 = c2 AND c2 * 10 = c3;
   c1  c2   c3
   --  --   --
 1 10 100 1000
 2 11 110 1100
 3 12 120 1200
 4  6  60  600
 5  7  70  700
 6  8  80  800
 7  9  90  900
(7 rows)
